File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/season/model/TvSeasonVideos.java
package dev.reuss.tmdb.domain.tv.season.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.video.VideosResponse;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Videos for a TMDB TV season.
 *
 * @param id      TMDB season id
 * @param results TV season videos
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeasonVideos(
        int id,
        List<TvSeasonVideo> results
) implements VideosResponse<TvSeasonVideo>, TmdbModel {

    public TvSeasonVideos {
        results = TmdbCollections.list(results);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/season/model/TvSeasonVideos.kt
package dev.reuss.tmdb.domain.tv.season.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.video.VideosResponse

/**
 * Videos for a TMDB TV season.
 *
 * @property id TMDB season id
 * @property results TV season videos
 */
@JvmRecord
data class TvSeasonVideos(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("results")
    override val results: List<TvSeasonVideo> = emptyList(),
) : VideosResponse<TvSeasonVideo>,
    TmdbModel
